File: ba1g.py
#if __name__ == "__main__":
#    with open("rosalind_ba1g.txt", "r") as f:
#        dnas = f.readlines()
#        for i in range(len(dnas)):
#            dnas[i] = dnas[i].strip()
#
#    k = len(dnas[0])
#    hamming_distance = 0
#
#    for base in range(0, k):
#        if dnas[0][base] != dnas[1][base]:
#            hamming_distance += 1
#
#    print("The hamming diatance is " + str(hamming_distance))

#with open("rosalind_ba1g.txt", "r") as f:
#    dna = f.readlines()
#    for i in range(len(dna)):
#        dna[i] = dna[i].strip()
#
#k = len(dna[0])
#hamming_distance = 0
#
#for j in range(0, k):
#    if dna[0][j] != dna[1][j]:
#        hamming_distance += 1
#
#print("The hamming distance is " + str(hamming_distance))
#with open("ba1g_output.txt", "w") as o:
#    o.write(str(hamming_distance))

with open("rosalind_ba1g.txt", "r") as f:
    dna = f.readlines()
    for i in range(len(dna)):
        dna[i] = dna[i].strip()
    print()

for j in range(0, 2):
    print(dna[j])

def calculate_hamming_distance (first_DNA, second_DNA):
    hamming_distance = 0
    k = len(first_DNA)
    for l in range(0, k):
        if first_DNA[l] != second_DNA[l]:
            hamming_distance += 1
    return hamming_distance

print (calculate_hamming_distance(dna[0], dna[1]))

File: ba1g.cpp
#include <iostream>
#include <iomanip>
#include <fstream>
#include <string>
#include <chrono>

using namespace std;

int main(void)
{
    string rosalindInputFileName;
    rosalindInputFileName.reserve(128);
    string rosalindOutputFileName;
    rosalindOutputFileName.reserve(128);

    cout << "Enter the rosalind input file's location and name:" << endl;
    cin >> setw(128) >> rosalindInputFileName;

    cout << "Enter the rosalind output file's location and name" << endl;
    cin >> setw(128) >> rosalindOutputFileName;

    cout << endl;

    auto start = std::chrono::steady_clock::now();

    string firstDNA;
    firstDNA.reserve(1024);
    string secondDNA;
    secondDNA.reserve(1024);

    ifstream rosalindInput;
    rosalindInput.open(rosalindInputFileName);
    if (!rosalindInput.is_open())
    {
        cerr << "Fail to open rosalind input file." << endl;
        return 1;
    }

    rosalindInput >> firstDNA >> secondDNA;
    cout << "The DNAs are:\n" << firstDNA << '\n' << secondDNA << '\n' << endl;

    rosalindInput.close();

    int hammingDistance = 0;
    const char* baseOfFirstDNA = firstDNA.c_str();
    const char* baseOfSecondDNA = secondDNA.c_str();
    while (*baseOfFirstDNA != '\0')
    {
        if (*baseOfFirstDNA != *baseOfSecondDNA)
        {
            hammingDistance++;
        }
        baseOfFirstDNA++;
        baseOfSecondDNA++;
    }
    baseOfFirstDNA = NULL;
    baseOfSecondDNA = NULL;

    cout << "The hamming distance is:\n" << hammingDistance << '\n' << endl;

    ofstream rosalindOutput;
    rosalindOutput.open(rosalindOutputFileName);
    rosalindOutput << hammingDistance << endl;

    rosalindOutput.close();
    cout << "Writing rosalind output is done." << '\n' << endl;

    auto end = std::chrono::steady_clock::now();
    std::chrono::duration<double> elapsed_seconds = end - start;
    std::cout << "elapsed time: " << elapsed_seconds.count() << "s\n";

    return 0;
}
